### PR TITLE
ASoC: SOF: dai-intel: add SOF_DAI_INTEL_SSP_CLKCTRL_MCLK/BCLK_ES bits

### PR DESCRIPTION
**Pull request by bardliao**
ASoC: SOF: dai-intel: add SOF_DAI_INTEL_SSP_CLKCTRL_MCLK/BCLK_ES bits

Add two clks_control bits. MCLK and/or BCLK will start during hw_params
and stop during hw_free if the corresponding bit is set.

### CODE REVIEW

File: include/uapi/sound/sof/abi.h [hunk 1]
```diff
@@ -27,7 +27,7 @@
 /* SOF ABI version major, minor and patch numbers */
 #define SOF_ABI_MAJOR 3
 #define SOF_ABI_MINOR 18
-#define SOF_ABI_PATCH 0
+#define SOF_ABI_PATCH 2
```
Comment (kv2019i) [inline include/uapi/sound/sof/abi.h]: NAK on this. This is an addition to the ABI so needs to bump MINOR version.

**Review by kv2019i — CHANGES_REQUESTED** [review]
This needs a ABI MINOR bump and needs to got through ABI process.

File: include/uapi/sound/sof/abi.h (hunk 1 above)
Comment (bardliao) [reply include/uapi/sound/sof/abi.h]: > NAK on this. This is an addition to the ABI so needs to bump MINOR version.

@kv2019i Thanks for the reminder, I will change it to MINOR version.

**Comment by bardliao** [comment]
> This needs a ABI MINOR bump and needs to got through ABI process.

Done, thanks.

**Review by plbossart — CHANGES_REQUESTED** [review]
I don't think we need a new token @bardliao, we already have a clk_control field in this structure.
https://github.com/thesofproject/linux/blob/dcea805219d0276916c6ef3266fa1c4d40eced55/include/sound/sof/dai-intel.h#L80

we added a 'Keep Alive' for Bclk and mclk, which IIRC meant 'remain active at all times from the moment the firmware boots'. We could add two fields for early_start for both MCLK and BCLK, and there would be no ABI change required.

**Comment by bardliao** [comment]
> I don't think we need a new token @bardliao, we already have a clk_control field in this structure.
> https://github.com/thesofproject/linux/blob/dcea805219d0276916c6ef3266fa1c4d40eced55/include/sound/sof/dai-intel.h#L80
> 
> we added a 'Keep Alive' for Bclk and mclk, which IIRC meant 'remain active at all times from the moment the firmware boots'. We could add two fields for early_start for both MCLK and BCLK, and there would be no ABI change required.

Thanks @plbossart I will do it.

**Review by plbossart — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Review by kv2019i — CHANGES_REQUESTED** [review]
Hmm, @plbossart and @bardliao .. even if we are using existing fields, taking new bits into use is still extending the interface, so I think this should follow the ABI process. We need sof/src/include/ipc/dai-intel.h change and a bump in ABI.

**Comment by plbossart** [comment]
> Hmm, @plbossart and @bardliao .. even if we are using existing fields, taking new bits into use is still extending the interface, so I think this should follow the ABI process. We need sof/src/include/ipc/dai-intel.h change and a bump in ABI.

Agree. let's wait a bit as well to see if we only provide support for BCLK and MCLK, there's an open as to whether FSYNC can be controlled separately.

**Comment by plbossart** [comment]
let's continue the work with https://github.com/thesofproject/linux/pull/2965

**Pull request closed without merging**